### PR TITLE
[Core] Fix get_token() issue in msi login and `expiresIn` key error in cloud shell login credentials for track 2 SDK related commands

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**  

**Testing Guide**  

- Test in Cloud shell as https://github.com/Azure/azure-cli/pull/14027#discussion_r448716269
- `az login --identity` and try track 2 extension

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/tests/test_profile.py [hunk 1]
```diff
@@ -645,7 +645,7 @@ def test_get_login_credentials_aux_tenants(self, mock_get_token, mock_read_cred_
                                                                      aux_tenants=[test_tenant_id2])
 
     @mock.patch('azure.cli.core._profile._load_tokens_from_file', autospec=True)
-    @mock.patch('msrestazure.azure_active_directory.MSIAuthentication', autospec=True)
+    @mock.patch('azure.cli.core._profile.MSIAuthenticationWrapper', autospec=True)
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/tests/test_profile.py:648]: Is this change necessary? Won't it be nice to have `MSIAuthenticationWrapper` also tested?

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 2]
```diff
@@ -21,6 +21,8 @@
 from azure.cli.core.util import get_file_json, in_cloud_console, open_page_in_browser, can_launch_browser,\
     is_windows, is_wsl
 from azure.cli.core.cloud import get_active_cloud, set_cloud_subscription
+from azure.core.credentials import AccessToken
+from msrestazure.azure_active_directory import MSIAuthentication
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_profile.py:25]: Please delay these imports as they will cause significant latency on client-side commands.

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_profile.py:25]: How about extracting `MSIAuthenticationWrapper` to `src/azure-cli-core/azure/cli/core/adal_authentication.py` and import it conditionally?

File: src/azure-cli-core/azure/cli/core/adal_authentication.py [hunk 3]
```diff
@@ -60,8 +61,10 @@ def _get_token(self):
     # This method is exposed for Azure Core.
     def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         _, token, full_token, _ = self._get_token()
-
-        return AccessToken(token, int(full_token['expiresIn'] + time.time()))
+        try:
+            return AccessToken(token, int(full_token['expiresIn'] + time.time()))
+        except KeyError:
+            return AccessToken(token, int(full_token['expires_on']))
```
Comment (jsntcy) [inline src/azure-cli-core/azure/cli/core/adal_authentication.py:67]: This logic is for cloud shell?

File: .gitignore [hunk 4]
```diff
@@ -65,6 +65,7 @@ src/build
 /doc/_build
 /doc/sphinx/_build
 /.vs/config/applicationhost.config
+.vscode/launch.json
```
Comment (jsntcy) [inline .gitignore:68]: No need to add this one as there is already .vscode/*

**Comment by yonzhan** [comment]
add to S172

File: src/azure-cli-core/azure/cli/core/adal_authentication.py (hunk 3 above)
Comment (Juliehzl) [reply src/azure-cli-core/azure/cli/core/adal_authentication.py:67]: sure

File: src/azure-cli-core/azure/cli/core/_profile.py (hunk 2 above)
Comment (Juliehzl) [reply src/azure-cli-core/azure/cli/core/_profile.py]: done

**Comment by jiasli** [comment]
Please specify what "login credentials issue" is.

File: src/azure-cli-core/azure/cli/core/tests/test_profile.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/tests/test_profile.py:648]: `mock.patch` by default will create a `MagicMock` instance which can't be inherited, maybe using `new=MSIAuthentication` may work? Haven't tested.

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 5]
```diff
@@ -22,6 +22,8 @@
     is_windows, is_wsl
 from azure.cli.core.cloud import get_active_cloud, set_cloud_subscription
 
+from .adal_authentication import MSIAuthenticationWrapper
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_profile.py:25]: Another problem with `mock.patch` is for this form of import, it won't work. After importing `MSIAuthenticationWrapper`, it becomes a local reference `azure.cli.core._profile.MSIAuthenticationWrapper`. See [Where to patch](https://docs.python.org/3/library/unittest.mock.html#where-to-patch).
